cerberus/client.py

```diff
@@ -327,30 +372,69 @@ def get_secrets(self, vault_path):
-                                   headers=self.HEADERS)
+        return self._get_secrets(vault_path, version)
 
-        self.throw_if_bad_response(secret_resp)
-
-        return secret_resp.json()
-
-    def get_secrets_data(self, vault_path):
+    def get_secrets_data(self, vault_path, version=None):
         """Return json secrets based on the vault_path
 
         Keyword arguments:
 
             vault_path (string) -- full path in the secret deposit box that contains the key
         """
-        secret_resp = requests.get(self.cerberus_url + '/v1/secret/' + vault_path,
-                                   headers=self.HEADERS)
+        return self._get_secrets(vault_path, version)['data']
 
-        self.throw_if_bad_response(secret_resp)
+    def get_secret_versions(self, vault_path, limit=None, offset=None):
+        """
+        Get verions of a particular secret key
+
+        vault_path -- full path to the key in the safety deposit box
+        limit -- Default(100), limits how many records to be returned from the api at once.
+        offset -- Default(0), used for pagination.  Will request records from the given offset.
+        """
+        # Set the normal defaults
+        if(limit is None):
+            limit = 100
 
-        return secret_resp.json()['data']
+        if(offset is None):
+            offset = 0
 
+        secret_resp = requests.get(str.join('', [self.cerberus_url, '/v1/secret-versions/', vault_path,
+                                   '?limit=', str(limit), '&offset=', str(offset)]), headers=self.HEADERS)
+        self.throw_if_bad_response(secret_resp)
+        return secret_resp.json()
+
+    def get_all_secret_version_ids(self, vault_path, limit=None):
+        """
+        Convience function that returns a generator that will paginate over the secret version ids
+        vault_path -- full path to the key in the safety deposit box
+        limit -- Default(100), limits how many records to be returned from the api at once.
+        """
+        offset = 0
+        versions = self.get_secret_versions(vault_path, limit, offset)
+        for summary in versions['secure_data_version_summaries']:
+            yield summary
+        while (versions['has_next']):
+            offset = versions['next_offset']
+            versions = self.get_secret_versions(vault_path, limit, offset)
+            for summary in versions['secure_data_version_summaries']:
+                yield summary
+
+    def get_all_secret_versions(self, vault_path, limit=None):
+        """
+        Convience function that returns a generator yielding the contents of secrets and their version info
+        vault_path -- full path to the key in the safety deposit box
+        limit -- Default(100), limits how many records to be returned from the api at once.
+        """
+        for secret in self.get_all_secret_version_ids(vault_path, limit):
+            yield {'secret': self.get_secrets_data(vault_path, version=secret['id']),
```

Comment:
While convenient to package the secret and it's version info into a dictionary, I wonder if I'm being too presumptuous about what potential users of this library will actually want.